src/js/model/blob/blob.js

```
@@ -0,0 +1,59 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.buildBlob = function(hexadecimal)
+{
+	// Defensive programming
+	if ( !(hexadecimal instanceof AMPHexConsumer) )
+	{
+		throw new ReferenceError("Did not receive an instance of AMPHexConsumer");
+	}
+
+	// Read the length of the blob
+	var blobLength;
+	var returnValue = {
+		type: "Blob",
+		value: []
+	};
+
+	try
+	{
+		blobLength = AMPParse.buildSdnv(hexadecimal);
+	}
+	catch (err)
+	{
+		err.nibblesConsumed = 0; // buildSdnv only throws errors when the hex is bad
+		throw err;
+	}
+
+	var nibblesConsumed = blobLength.nibblesConsumed;
+	blobLength = blobLength.returnValue.value;
+
+	console.log("blobLength: " + blobLength);
+
+	// Repeatedly read in bytes
+	for (var i = 0; i < blobLength; i++)
+	{
+		try
+		{
+			if (hexadecimal.isEmpty)
+			{
+				throw new RangeError("There were fewer bytes than stated by the length SDNV.");
+			}
+
```

Only question I have is, should we update the AMPSetup.js to throw an error when we try to call consume when the remaining hex is empty and then handle adding the more detailed message in catch block.  If not feel free to merge

---

I agree the handling you have in SDNV and String makes sense,  but I still think we should throw a range exception if we call consume on an empty string.  If we do somehow get in  a case where we call consume on empty string I'd think we'd want the same exception as if we called consume on string that is too short

---

Fair enough. Make an issue for it and I'll throw it in on another branch